tests/test_modeling.py

```diff
@@ -0,0 +1,223 @@
+    adapter = MCPAdapt(
+        StdioServerParameters(
+            command="uv",
+            args=["run", "python", "-c", json_schema_array_type_server_script],
+        ),
+        CrewAIAdapter(),
+    )
+
+    # If we can't access private attributes, skip the test
+    if not hasattr(adapter, "_sessions"):
+        print(
+            "\nSkipping schema inspection - cannot access adapter internal structure",
+            file=sys.stderr,
+        )
+        return
+
+    try:
+        # Enter the context to initialize connection
+        with adapter as _:  # Use underscore to indicate unused variable
+            # Now examine the internal state of the adapter
+            sessions = adapter._sessions
+
+            if sessions:
+                for session_idx, session in enumerate(sessions):
+                    if hasattr(session, "_mcp_server") and hasattr(
+                        session._mcp_server, "tools"
+                    ):
+                        mcp_server = session._mcp_server
+                        print(
+                            f"\n\n=== Session {session_idx} Tools ===", file=sys.stderr
+                        )
+
+                        for tool_idx, tool in enumerate(mcp_server.tools):
+                            print(
+                                f"\nTool {tool_idx}: {getattr(tool, 'name', '[NO NAME]')}",
+                                file=sys.stderr,
+                            )
+
+                            # Try to inspect the tool object
+                            try:
+                                print(
+                                    f"Tool attrs: {dir(tool)[:10]}...", file=sys.stderr
+                                )
+                            except Exception as e:
+                                print(
+                                    f"Could not inspect tool attributes: {str(e)}",
+                                    file=sys.stderr,
+                                )
+
+                            # Extract and print available schemas
+                            for schema_attr in [
+                                "param_schema",
+                                "inputSchema",
+                                "schema",
+                            ]:
+                                if hasattr(tool, schema_attr) and getattr(
+                                    tool, schema_attr
+                                ):
+                                    print(f"{schema_attr}:", file=sys.stderr)
+                                    print(
+                                        json.dumps(
+                                            getattr(tool, schema_attr), indent=2
+                                        ),
+                                        file=sys.stderr,
+                                    )
+    except Exception as e:
+        print(f"Error during schema inspection: {e}", file=sys.stderr)
+
+
+def test_direct_modeling_with_list_type():
```

Comment:
this test could be nice but I would remove the verbose try / except explaining the type error and just keep it to the essential